Week3/homework/src/App.js

```diff
@@ -0,0 +1,131 @@
+'use strict';
+//sort?? ARIA il
+/* global Util, Repository, Contributor */
+
+class App {
+  constructor(url) {
+    this.initialize(url);
+  }
+
+  /**
+   * Initialization
+   * @param {string} url The GitHub URL for obtaining the organization's repositories.
+   */
+  async initialize(url) {
+    // Add code here to initialize your app
+    // 1. Create the fixed HTML elements of your page
+    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
+
+    const root = document.getElementById('root');
+    const divRepositoryContainer = Util.createAndAppend('div', root, { id: 'divRepositoryContainer' });
+    Util.createAndAppend('div', root, { class: 'clsRepositoryDetails', id: 'container' });
+
+    // ...
+
+    try {
+      // ...
+      const repos = await Util.fetchJSON(url);
+      this.repos = repos.map(repo => new Repository(repo));
+      const sortedData = this.sortData(repos, 'name');
+      this.sortedData = sortedData.map(sData => new Repository(sData));//Hoe kan ik bind to class?
+
+      Util.createAndAppend('p', divRepositoryContainer, { text: 'HYF Repositories', id: 'repositoryLabel' });
+      const select = Util.createAndAppend('select', divRepositoryContainer, { id: 'selectRepositories' });
+
+      sortedData.forEach(elem => {
+        Util.createAndAppend('option', select, { text: elem.name, value: elem.id });
+      });
+
+      select.addEventListener('change', e => this.onSelectionChange(e.target.value));
+
+      this.fetchContributorsAndRender(select.selectedIndex);
+
+    } catch (error) {
+      this.renderError(error);
+    }
+
+  }
+
+  async onSelectionChange(url) {
+    const repoDetailContainer = document.getElementById('container');
+    this.clearContainer(repoDetailContainer);
+    const select = document.getElementById('selectRepositories');
+    if (url === '') {
+      return;
+    }
+    try {
+      this.fetchContributorsAndRender(select.selectedIndex);
+    } catch (error) {
+      this.renderError(error);
+    }
+
+  }
+
+  sortData(data, key) {
```

Comment:
I already made a comment about not needing this function at all in a previous review.